contracts/PriceOracle/PriceOracle.sol

```diff
@@ -110,34 +103,30 @@ contract PriceOracle is Ownable2Step {
         // Initialize the aggregator to read the price from
         AggregatorV3Interface priceFeed = AggregatorV3Interface(config.priceFeed);
         // Retrieve decimals from feed for formatting
-        uint8 decimals = priceFeed.decimals();
-        // Fail safe check since this is an unrealistic scenario. The number of digits that the int256 answer can have naturally limits decimals
-        // but this protects against an erroneous uint8 values
-        if (decimals > MAX_DECIMALS) revert DecimalsTooLarge(decimals);
+        uint8 feedDecimals = priceFeed.decimals();
         // Retrieve price from feed
         (
             /* uint80 roundID */,
             int256 answer,
-            /*uint startedAt*/,
-            /*uint timeStamp*/,
+            /*uint256 startedAt*/,
+            /*uint256 updatedAt*/,
             /*uint80 answeredInRound*/
         ) = priceFeed.latestRoundData();
-        if (answer < 0) revert NegativePrice(answer);
+        // Invalid price returned by feed. Comptroller expects 0 price on error.
+        if (answer <= 0) return 0;
         uint256 price = uint256(answer);
 
         // Number of decimals determines whether the price needs to be multiplied or divided for scaling
         // Handle the 2 scenarios separately to ensure a non-fractional scale value
-        if (decimals <= PRICE_SCALE) {
+        if (feedDecimals + config.underlyingAssetDecimals <= PRICE_SCALE) {
             // Decimals is always >=0 so the scale max value is 1e36 here and not at risk of overflowing
-            uint256 scale = 10 ** (PRICE_SCALE - decimals);
-            return FullMath.mulDiv(price, scale, config.baseUnit);
+            uint256 scale = 10 ** (PRICE_SCALE - feedDecimals - config.underlyingAssetDecimals);
```

Comment:
Will there ever be a case this subtraction becomes negative?

Reply:
I don't think so. The if clause `feedDecimals + config.underlyingAssetDecimals <= PRICE_SCALE` should ensure that it doesn't.